### PR TITLE
String content emoji

### LINKED ISSUE #10573: StringHtmlContent omitting characters when encoding
### Describe the bug
When a `StringHtmlContent` writes a string containing a surrogate pair, it omits the immediate following single non-supplementary character.

### To Reproduce
1. Using this version of ASP.NET Core 2.2.5
2. Run this code
```
    var tearsOfJoy2 = new StringHtmlContent("😂2");
    using (var stringWriter = new StringWriter())
    {
        tearsOfJoy2.WriteTo((TextWriter) stringWriter, HtmlEncoder.Default);
        var encoded = stringWriter.ToString();
    }
```

### Expected behaviour
`encoded == "&#x1F602;2"`

### Actual behaviour
`encoded == "&#x1F602;"`

### Additional context
I have a feeling that the issue is within `System.Text.Encodings.Web.TextEncoder`.
If the the emoji is followed by 2 characters, like `new StringHtmlContent("😂22")`, it seems to be working correctly.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Tests the fix of https://github.com/aspnet/AspNetCore/issues/10573.

Once this test passes we'll know that the fix from corefx has made it into our build, we can merge this and close https://github.com/aspnet/AspNetCore/issues/10573. Obviously you'll have to rebase it to pick up new versions and run the tests again.

### CODE REVIEW

File: src/Mvc/Mvc.ViewFeatures/test/StringHtmlContentTest.cs [hunk 1]
```diff
@@ -22,5 +23,19 @@ public void WriteTo_WritesContent()
...
         }
+
+        [Fact]
+        public void Emoji_EncodedCorrectly()
+        {
+            // Arrange & Act
+            var tearsOfJoy = new StringHtmlContent("😂2");
+
+            // Assert
+            using (var stringWritter = new StringWriter())
```
Comment (gfoidl) [inline src/Mvc/Mvc.ViewFeatures/test/StringHtmlContentTest.cs]: Typo: `stringWriter` (one `t` too much).

**Comment by ryanbrandenburg** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=238924

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2293
https://github.com/aspnet/AspNetCore-Internal/issues/2577

**Review by jkotalik — APPROVED** [review]
Sure.

File: src/Mvc/Mvc.ViewFeatures/test/StringHtmlContentTest.cs [hunk 2]
```diff
@@ -22,5 +23,19 @@ public void WriteTo_WritesContent()
...
+        public void Emoji_EncodedCorrectly()
+        {
+            // Arrange & Act
+            var tearsOfJoy = new StringHtmlContent("😂2");
+
+            // Assert
+            using (var stringWriter = new StringWriter())
+            {
+                tearsOfJoy.WriteTo(stringWriter, HtmlEncoder.Default);
+                Assert.Equal("&#x1f602;2", stringWriter.ToString(), ignoreCase: true);
```
Comment (jkotalik) [inline src/Mvc/Mvc.ViewFeatures/test/StringHtmlContentTest.cs]: Why do you need to pass in ignoreCase?

**Pull request merged**